linetools/spectra/io.py

```diff
@@ -165,26 +163,37 @@ def readspec(specfil, inflg=None, efil=None, verbose=False, multi_ivar=False,
         print('Not sure what has been input.  Send to JXP.')
         return
 
-    if np.any(np.isnan(xspec1d.dispersion)):
+
+    # Generate, as needed
+    if 'xspec1d' not in locals():
+        # Give Ang as default
+        if not hasattr(wave, 'unit'):
+            uwave = u.Quantity(wave, unit=u.AA)
+        elif wave.unit is None:
+            uwave = u.Quantity(wave, unit=u.AA)
+        else:
+            uwave = u.Quantity(wave)
+        xspec1d = XSpectrum1D.from_tuple((wave, fx, sig, None))
+
```

Comment:
I believe it was because of a bad merge (I fear I didn't pull your refactor
until I started fiddling about).

Am commenting this block out now and if things are ok will kill it altogether.